ecommerce/courses/publishers.py

```
@@ -37,31 +39,24 @@ def serialize_seat_for_commerce_api(self, seat):
 
     def _publish_creditcourse(self, course_id, access_token):
         """Creates or updates a CreditCourse object on the LMS."""
-        url = get_lms_url('api/credit/v1/courses/')
-
         data = {
             'course_key': course_id,
             'enabled': True
         }
 
-        headers = {
-            'Content-Type': 'application/json',
-            'Authorization': 'Bearer ' + access_token
-        }
+        api = EdxRestApiClient(
+            get_lms_url('api/credit/v1/'),
+            oauth_access_token=access_token,
+            timeout=self.timeout
+        )
 
-        kwargs = {
-            'url': url,
-            'data': json.dumps(data),
-            'headers': headers,
-            'timeout': self.timeout
-        }
-        response = requests.post(**kwargs)
-        if response.status_code == 400:
-            # The CreditCourse already exists. Try updating it.
-            kwargs['url'] += course_id.strip('/') + '/'
-            response = requests.put(**kwargs)
+        try:
+            api.courses(course_id).put(data)
+            return True
+        except (SlumberBaseException, Timeout):
+            logger.exception('Failed to create or update CreditCourse [%s].', course_id)
 
-        return response
+        return False
```

Returning `True` or `False` to indicate an failure state seems suboptimal to me. Would it be better to just let the exceptions raise, and then catch them in the calling method?